# Results comparison for MIST filtering of model B

In this notebook, I will compare the results of the different models relative to how the data was filtered (only the MIST dataset).

I will first look at how the filter on one value affected the other (if it didn't affect it, I will not write it) and then see how it affected the results of each model.

In [1]:
from ML.utils.utils import *
from physics.Iso_data_handler import Iso_data_handler
from ML.utils.Data_preparator import Data_preparator

In [2]:
pre_path = "../../../../../../"
path = pre_path + "results/model_B/default_models"

In [3]:
compare_metrics(path, value_rounding=3,
                output_parameters=["T_eff"], # output_parameters=["radius"],
                model_names=["random_forest"],
                physical_models=["MIST"],
                data_filters = ["Base", "mass_smaller_3", "mass_smaller_14", "mass_smaller_30"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter          output_parameter                  
random_forest MIST           Base            T_eff             3.345 - 5.379   
                             mass_smaller_3  T_eff             3.345 - 5.214   
                             mass_smaller_14 T_eff             3.345 - 5.214   
                             mass_smaller_30 T_eff             3.345 - 5.354   

                                                                 RVE   RMSE  \
model         physical_model filter          output_parameter                 
random_forest MIST           Base            T_eff             0.966  0.064   
                             mass_smaller_3  T_eff             0.958  0.041   
                             mass_smaller_14 T_eff             0.967  0.053   
                             mass_smaller_30 T_eff             0.964  0.061   

                                                                 MAE  MedAE  \
model         physical_model filter          output_parameter                 
random_forest MIST           Base            T_eff             0.017  0.001   
                             mass_smaller_3  T_eff             0.015  0.001   
                             mass_smaller_14 T_eff             0.014  0.001   
                             mass_smaller_30 T_eff             0.015  0.001   

                                                                CORR  MAX_ER  \
model         physical_model filter          output_parameter                  
random_forest MIST           Base            T_eff             0.983   0.982   
                             mass_smaller_3  T_eff             0.979   1.206   
                             mass_smaller_14 T_eff             0.983   1.587   
                             mass_smaller_30 T_eff             0.982   1.609   

                                                                                                               Percentiles  \
model         physical_model filter          output_parameter                                                                
random_forest MIST           Base            T_eff              75 : 0.00404 / 90 : 0.03266 / 95 : 0.09054 / 99 : 0.32479    
                             mass_smaller_3  T_eff              75 : 0.00586 / 90 : 0.04883 / 95 : 0.09348 / 99 : 0.18479    
                             mass_smaller_14 T_eff              75 : 0.00391 / 90 : 0.02873 / 95 : 0.07799 / 99 : 0.20813    
                             mass_smaller_30 T_eff              75 : 0.00382 / 90 : 0.02957 / 95 : 0.08364 / 99 : 0.28184    

                                                                  time  
model         physical_model filter          output_parameter           
random_forest MIST           Base            T_eff             243.748  
                             mass_smaller_3  T_eff             119.180  
                             mass_smaller_14 T_eff             228.025  
                             mass_smaller_30 T_eff             258.963

Results for the phase category with a value of 0.0


value range  \
model         physical_model filter          output_parameter                  
random_forest MIST           Base            T_eff             3.345 - 4.903   
                             mass_smaller_3  T_eff             3.345 - 4.301   
                             mass_smaller_14 T_eff             3.345 - 4.634   
                             mass_smaller_30 T_eff             3.345 - 4.898   

                                                                 RVE   RMSE  \
model         physical_model filter          output_parameter                 
random_forest MIST           Base            T_eff             0.974  0.057   
                             mass_smaller_3  T_eff             0.985  0.024   
                             mass_smaller_14 T_eff             0.981  0.042   
                             mass_smaller_30 T_eff             0.972  0.054   

                                                                 MAE  MedAE  \
model         physical_model filter          output_parameter                 
random_forest MIST           Base            T_eff             0.010  0.001   
                             mass_smaller_3  T_eff             0.005  0.000   
                             mass_smaller_14 T_eff             0.006  0.001   
                             mass_smaller_30 T_eff             0.009  0.001   

                                                                CORR  MAX_ER  \
model         physical_model filter          output_parameter                  
random_forest MIST           Base            T_eff             0.987   0.979   
                             mass_smaller_3  T_eff             0.993   1.206   
                             mass_smaller_14 T_eff             0.990   1.587   
                             mass_smaller_30 T_eff             0.986   1.609   

                                                                                                               Percentiles  \
model         physical_model filter          output_parameter                                                                
random_forest MIST           Base            T_eff               75 : 0.00149 / 90 : 0.00507 / 95 : 0.02312 / 99 : 0.3164    
                             mass_smaller_3  T_eff               75 : 0.00129 / 90 : 0.00368 / 95 : 0.0103 / 99 : 0.13197    
                             mass_smaller_14 T_eff              75 : 0.00129 / 90 : 0.00349 / 95 : 0.00889 / 99 : 0.14038    
                             mass_smaller_30 T_eff               75 : 0.00137 / 90 : 0.0042 / 95 : 0.01532 / 99 : 0.25128    

                                                                  time  
model         physical_model filter          output_parameter           
random_forest MIST           Base            T_eff             243.748  
                             mass_smaller_3  T_eff             119.180  
                             mass_smaller_14 T_eff             228.025  
                             mass_smaller_30 T_eff             258.963

Results for the phase category with a value of 2.0


value range  \
model         physical_model filter          output_parameter                  
random_forest MIST           Base            T_eff             3.425 - 4.711   
                             mass_smaller_3  T_eff             3.425 - 4.277   
                             mass_smaller_14 T_eff             3.425 - 4.588   
                             mass_smaller_30 T_eff             3.425 - 4.675   

                                                                 RVE   RMSE  \
model         physical_model filter          output_parameter                 
random_forest MIST           Base            T_eff             0.975  0.050   
                             mass_smaller_3  T_eff             0.844  0.057   
                             mass_smaller_14 T_eff             0.956  0.052   
                             mass_smaller_30 T_eff             0.970  0.050   

                                                                 MAE  MedAE  \
model         physical_model filter          output_parameter                 
random_forest MIST           Base            T_eff             0.020  0.002   
                             mass_smaller_3  T_eff             0.033  0.008   
                             mass_smaller_14 T_eff             0.022  0.003   
                             mass_smaller_30 T_eff             0.021  0.002   

                                                                CORR  MAX_ER  \
model         physical_model filter          output_parameter                  
random_forest MIST           Base            T_eff             0.987   0.704   
                             mass_smaller_3  T_eff             0.920   0.300   
                             mass_smaller_14 T_eff             0.978   0.697   
                             mass_smaller_30 T_eff             0.985   0.713   

                                                                                                               Percentiles  \
model         physical_model filter          output_parameter                                                                
random_forest MIST           Base            T_eff              75 : 0.01116 / 90 : 0.06942 / 95 : 0.11179 / 99 : 0.20013    
                             mass_smaller_3  T_eff               75 : 0.0495 / 90 : 0.10251 / 95 : 0.13649 / 99 : 0.19511    
                             mass_smaller_14 T_eff              75 : 0.01776 / 90 : 0.07706 / 95 : 0.11769 / 99 : 0.19591    
                             mass_smaller_30 T_eff               75 : 0.01362 / 90 : 0.0722 / 95 : 0.11353 / 99 : 0.19463    

                                                                  time  
model         physical_model filter          output_parameter           
random_forest MIST           Base            T_eff             243.748  
                             mass_smaller_3  T_eff             119.180  
                             mass_smaller_14 T_eff             228.025  
                             mass_smaller_30 T_eff             258.963

Results for the phase category with a value of 3.0


value range  \
model         physical_model filter          output_parameter                  
random_forest MIST           Base            T_eff             3.352 - 5.379   
                             mass_smaller_3  T_eff             3.352 - 5.214   
                             mass_smaller_14 T_eff             3.352 - 5.214   
                             mass_smaller_30 T_eff             3.352 - 5.354   

                                                                 RVE   RMSE  \
model         physical_model filter          output_parameter                 
random_forest MIST           Base            T_eff             0.923  0.091   
                             mass_smaller_3  T_eff             0.954  0.053   
                             mass_smaller_14 T_eff             0.917  0.077   
                             mass_smaller_30 T_eff             0.916  0.088   

                                                                 MAE  MedAE  \
model         physical_model filter          output_parameter                 
random_forest MIST           Base            T_eff             0.029  0.003   
                             mass_smaller_3  T_eff             0.021  0.003   
                             mass_smaller_14 T_eff             0.023  0.003   
                             mass_smaller_30 T_eff             0.026  0.003   

                                                                CORR  MAX_ER  \
model         physical_model filter          output_parameter                  
random_forest MIST           Base            T_eff             0.961   0.982   
                             mass_smaller_3  T_eff             0.977   1.030   
                             mass_smaller_14 T_eff             0.958   0.885   
                             mass_smaller_30 T_eff             0.958   0.991   

                                                                                                               Percentiles  \
model         physical_model filter          output_parameter                                                                
random_forest MIST           Base            T_eff               75 : 0.0133 / 90 : 0.06569 / 95 : 0.15397 / 99 : 0.48948    
                             mass_smaller_3  T_eff              75 : 0.01446 / 90 : 0.05848 / 95 : 0.10576 / 99 : 0.25125    
                             mass_smaller_14 T_eff              75 : 0.01185 / 90 : 0.04991 / 95 : 0.10873 / 99 : 0.40663    
                             mass_smaller_30 T_eff              75 : 0.01159 / 90 : 0.05354 / 95 : 0.12828 / 99 : 0.49788    

                                                                  time  
model         physical_model filter          output_parameter           
random_forest MIST           Base            T_eff             243.748  
                             mass_smaller_3  T_eff             119.180  
                             mass_smaller_14 T_eff             228.025  
                             mass_smaller_30 T_eff             258.963

In [4]:
compare_metrics(path, value_rounding=3,
                output_parameters=["T_eff", "radius"], # output_parameters=["radius"],
                physical_models=["MIST"],
                data_filters=["Base"])

Results for the Global category with a value of 1.0


value range  \
model             physical_model filter output_parameter                  
decision_tree     MIST           Base   T_eff             3.345 - 5.379   
                                        radius            -1.861 - 3.13   
KNN               MIST           Base   T_eff             3.345 - 5.379   
                                        radius            -1.861 - 3.13   
linear_regression MIST           Base   T_eff             3.345 - 5.379   
                                        radius            -1.861 - 3.13   
mlp               MIST           Base   T_eff             3.345 - 5.379   
                                        radius            -1.861 - 3.13   
random_forest     MIST           Base   T_eff             3.345 - 5.379   
                                        radius            -1.861 - 3.13   
xgboost           MIST           Base   T_eff             3.345 - 5.379   
                                        radius            -1.861 - 3.13   

                                                            RVE   RMSE    MAE  \
model             physical_model filter output_parameter                        
decision_tree     MIST           Base   T_eff             0.949  0.079  0.019   
                                        radius            0.811  0.324  0.090   
KNN               MIST           Base   T_eff             0.967  0.063  0.019   
                                        radius            0.903  0.232  0.078   
linear_regression MIST           Base   T_eff             0.697  0.191  0.147   
                                        radius            0.193  0.669  0.551   
mlp               MIST           Base   T_eff             0.688  0.195  0.132   
                                        radius            0.559  0.499  0.327   
random_forest     MIST           Base   T_eff             0.966  0.064  0.017   
                                        radius            0.875  0.263  0.079   
xgboost           MIST           Base   T_eff             0.896  0.112  0.070   
                                        radius            0.817  0.318  0.203   

                                                          MedAE   CORR  \
model             physical_model filter output_parameter                 
decision_tree     MIST           Base   T_eff             0.002  0.974   
                                        radius            0.005  0.906   
KNN               MIST           Base   T_eff             0.002  0.983   
                                        radius            0.004  0.951   
linear_regression MIST           Base   T_eff             0.131  0.835   
                                        radius            0.476  0.440   
mlp               MIST           Base   T_eff             0.086  0.841   
                                        radius            0.183  0.760   
random_forest     MIST           Base   T_eff             0.001  0.983   
                                        radius            0.002  0.936   
xgboost           MIST           Base   T_eff             0.035  0.947   
                                        radius            0.101  0.904   

                                                          MAX_ER  \
model             physical_model filter output_parameter           
decision_tree     MIST           Base   T_eff              1.376   
                                        radius             3.855   
KNN               MIST           Base   T_eff              1.625   
                                        radius             3.017   
linear_regression MIST           Base   T_eff              1.529   
                                        radius             2.690   
mlp               MIST           Base   T_eff              1.376   
                                        radius             2.923   
random_forest     MIST           Base   T_eff              0.982   
                                        radius             3.415   
xgboost           MIS

Results for the phase category with a value of 0.0


value range  \
model             physical_model filter output_parameter                   
decision_tree     MIST           Base   T_eff              3.345 - 4.903   
                                        radius            -0.997 - 3.083   
KNN               MIST           Base   T_eff              3.345 - 4.903   
                                        radius            -0.997 - 3.083   
linear_regression MIST           Base   T_eff              3.345 - 4.903   
                                        radius            -0.997 - 3.083   
mlp               MIST           Base   T_eff              3.345 - 4.903   
                                        radius            -0.997 - 3.083   
random_forest     MIST           Base   T_eff              3.345 - 4.903   
                                        radius            -0.997 - 3.083   
xgboost           MIST           Base   T_eff              3.345 - 4.903   
                                        radius            -0.997 - 3.083   

                                                            RVE   RMSE    MAE  \
model             physical_model filter output_parameter                        
decision_tree     MIST           Base   T_eff             0.959  0.072  0.012   
                                        radius            0.833  0.235  0.039   
KNN               MIST           Base   T_eff             0.968  0.063  0.016   
                                        radius            0.896  0.187  0.046   
linear_regression MIST           Base   T_eff             0.753  0.177  0.151   
                                        radius            0.454  0.621  0.520   
mlp               MIST           Base   T_eff             0.828  0.149  0.102   
                                        radius            0.678  0.341  0.213   
random_forest     MIST           Base   T_eff             0.974  0.057  0.010   
                                        radius            0.895  0.187  0.035   
xgboost           MIST           Base   T_eff             0.939  0.093  0.051   
                                        radius            0.833  0.254  0.142   

                                                          MedAE   CORR  \
model             physical_model filter output_parameter                 
decision_tree     MIST           Base   T_eff             0.001  0.980   
                                        radius            0.003  0.925   
KNN               MIST           Base   T_eff             0.001  0.984   
                                        radius            0.002  0.954   
linear_regression MIST           Base   T_eff             0.150  0.868   
                                        radius            0.474  0.680   
mlp               MIST           Base   T_eff             0.069  0.914   
                                        radius            0.083  0.871   
random_forest     MIST           Base   T_eff             0.001  0.987   
                                        radius            0.001  0.951   
xgboost           MIST           Base   T_eff             0.017  0.969   
                                        radius            0.040  0.932   

                                                          MAX_ER  \
model             physical_model filter output_parameter           
decision_tree     MIST           Base   T_eff              1.376   
                                        radius             2.836   
KNN               MIST           Base   T_eff              1.625   
                                        radius             2.792   
linear_regression MIST           Base   T_eff              0.710   
                                        radius             2.031   
mlp               MIST           Base   T_eff              1.084   
                                        radius             2.401   
random_forest     MIST           Base   T_eff              0.979   
                                        radius             2.800   
xgboost 

Results for the phase category with a value of 2.0


value range  \
model             physical_model filter output_parameter                  
decision_tree     MIST           Base   T_eff             3.425 - 4.711   
                                        radius              0.092 - 2.9   
KNN               MIST           Base   T_eff             3.425 - 4.711   
                                        radius              0.092 - 2.9   
linear_regression MIST           Base   T_eff             3.425 - 4.711   
                                        radius              0.092 - 2.9   
mlp               MIST           Base   T_eff             3.425 - 4.711   
                                        radius              0.092 - 2.9   
random_forest     MIST           Base   T_eff             3.425 - 4.711   
                                        radius              0.092 - 2.9   
xgboost           MIST           Base   T_eff             3.425 - 4.711   
                                        radius              0.092 - 2.9   

                                                            RVE   RMSE    MAE  \
model             physical_model filter output_parameter                        
decision_tree     MIST           Base   T_eff             0.959  0.064  0.024   
                                        radius            0.235  0.430  0.176   
KNN               MIST           Base   T_eff             0.983  0.041  0.017   
                                        radius            0.648  0.292  0.125   
linear_regression MIST           Base   T_eff             0.776  0.150  0.120   
                                        radius            0.303  0.589  0.468   
mlp               MIST           Base   T_eff             0.595  0.209  0.152   
                                        radius           -0.207  0.563  0.424   
random_forest     MIST           Base   T_eff             0.975  0.050  0.020   
                                        radius            0.472  0.357  0.151   
xgboost           MIST           Base   T_eff             0.875  0.114  0.085   
                                        radius            0.497  0.352  0.265   

                                                          MedAE   CORR  \
model             physical_model filter output_parameter                 
decision_tree     MIST           Base   T_eff             0.003  0.980   
                                        radius            0.008  0.625   
KNN               MIST           Base   T_eff             0.002  0.992   
                                        radius            0.005  0.810   
linear_regression MIST           Base   T_eff             0.106  0.882   
                                        radius            0.367  0.564   
mlp               MIST           Base   T_eff             0.110  0.801   
                                        radius            0.330  0.425   
random_forest     MIST           Base   T_eff             0.002  0.987   
                                        radius            0.005  0.721   
xgboost           MIST           Base   T_eff             0.064  0.936   
                                        radius            0.199  0.706   

                                                          MAX_ER  \
model             physical_model filter output_parameter           
decision_tree     MIST           Base   T_eff              0.830   
                                        radius             2.319   
KNN               MIST           Base   T_eff              0.713   
                                        radius             2.188   
linear_regression MIST           Base   T_eff              0.626   
                                        radius             1.812   
mlp               MIST           Base   T_eff              1.039   
                                        radius             2.549   
random_forest     MIST           Base   T_eff              0.704   
                                        radius             2.318   
xgboost           MIS

Results for the phase category with a value of 3.0


value range  \
model             physical_model filter output_parameter                  
decision_tree     MIST           Base   T_eff             3.352 - 5.379   
                                        radius            -1.861 - 3.13   
KNN               MIST           Base   T_eff             3.352 - 5.379   
                                        radius            -1.861 - 3.13   
linear_regression MIST           Base   T_eff             3.352 - 5.379   
                                        radius            -1.861 - 3.13   
mlp               MIST           Base   T_eff             3.352 - 5.379   
                                        radius            -1.861 - 3.13   
random_forest     MIST           Base   T_eff             3.352 - 5.379   
                                        radius            -1.861 - 3.13   
xgboost           MIST           Base   T_eff             3.352 - 5.379   
                                        radius            -1.861 - 3.13   

                                                            RVE   RMSE    MAE  \
model             physical_model filter output_parameter                        
decision_tree     MIST           Base   T_eff             0.888  0.110  0.033   
                                        radius            0.696  0.363  0.114   
KNN               MIST           Base   T_eff             0.934  0.085  0.030   
                                        radius            0.859  0.249  0.098   
linear_regression MIST           Base   T_eff             0.383  0.265  0.175   
                                        radius            0.287  0.869  0.748   
mlp               MIST           Base   T_eff             0.438  0.273  0.186   
                                        radius            0.197  0.720  0.509   
random_forest     MIST           Base   T_eff             0.923  0.091  0.029   
                                        radius            0.808  0.289  0.099   
xgboost           MIST           Base   T_eff             0.828  0.150  0.102   
                                        radius            0.706  0.413  0.286   

                                                          MedAE   CORR  \
model             physical_model filter output_parameter                 
decision_tree     MIST           Base   T_eff             0.004  0.945   
                                        radius            0.009  0.845   
KNN               MIST           Base   T_eff             0.004  0.967   
                                        radius            0.013  0.927   
linear_regression MIST           Base   T_eff             0.112  0.648   
                                        radius            0.597  0.549   
mlp               MIST           Base   T_eff             0.109  0.721   
                                        radius            0.305  0.536   
random_forest     MIST           Base   T_eff             0.003  0.961   
                                        radius            0.009  0.899   
xgboost           MIST           Base   T_eff             0.065  0.910   
                                        radius            0.179  0.848   

                                                          MAX_ER  \
model             physical_model filter output_parameter           
decision_tree     MIST           Base   T_eff              1.225   
                                        radius             3.855   
KNN               MIST           Base   T_eff              0.930   
                                        radius             3.017   
linear_regression MIST           Base   T_eff              1.529   
                                        radius             2.690   
mlp               MIST           Base   T_eff              1.376   
                                        radius             2.923   
random_forest     MIST           Base   T_eff              0.982   
                                        radius             3.415   
xgboost           MIS

## Getting the ranges for the values

In [5]:
iso_handler = Iso_data_handler("C:/Users/antoi/Code/unif/MA2/thesis/data/MIST_v1.2_vvcrit0.0_basic_isos/", 
                                    ['log10_isochrone_age_yr', 'log_Teff', 'log_g', 'phase', 'metallicity', 'star_mass', 'log_R', 'log_L'], 
                                    "MIST", reclassify=True)

iso_df = iso_handler.get_isochrone_dataframe()
display(iso_df)

Reading MIST dataframe from csv file...


,log10_isochrone_age_yr,log_Teff,log_g,phase,metallicity,star_mass,log_R,log_L
0,5.0,3.486221,3.131342,-1.0,-0.25,0.100000,0.153402,-0.793667
1,5.0,3.487362,3.126808,-1.0,-0.25,0.102645,0.160326,-0.775254
2,5.0,3.489243,3.119367,-1.0,-0.25,0.107039,0.171785,-0.744810
3,5.0,3.491102,3.112165,-1.0,-0.25,0.111419,0.183099,-0.714749
4,5.0,3.492937,3.105143,-1.0,-0.25,0.115789,0.194305,-0.684996
...,...,...,...,...,...,...,...,...
1467117,10.3,4.402490,7.777159,6.0,0.50,0.532726,-1.806255,-1.047903
1467118,10.3,4.387132,7.783242,6.0,0.50,0.532730,-1.809295,-1.115415
1467119,10.3,4.371789,7.789130,6.0,0.50,0.532735,-1.812237,-1.182672
1467120,10.3,4.356480,7.794844,6.0,0.50,0.532741,-1.815091,-1.249617


In [7]:
phase_filtered_iso_df = Data_preparator.filter_data(iso_df, {'phase':[0, 2, 3, 4, 5]})
print_all_uniques(phase_filtered_iso_df)

log10_isochrone_age_yr  : Range : 5.0 - 10.3, Mean : 8.4794, Median : 8.6

log_Teff  : Range : 3.34 - 5.5871, Mean : 3.8475, Median : 3.6808

log_g  : Range : -1.14 - 8.6121, Mean : 2.1347, Median : 2.0681

Values in phase column : 0.0, 2.0, 3.0, 4.0, 5.0 

Values in metallicity column : -0.25, -0.5, -0.75, -1.0, -1.25, -1.5, -1.75, -2.0, -2.5, -3.0, -3.5, -4.0, 0.0, 0.25, 0.5 

star_mass  : Range : 0.1 - 298.5448, Mean : 7.3915, Median : 2.0304

log_R  : Range : -2.09 - 3.1298, Mean : 1.3543, Median : 1.4604

log_L  : Range : -3.33 - 7.069, Mean : 3.0533, Median : 3.4128



## Base vs Base_old

| Parameter      | Base                                                    | Base_old                                               |
| -------------  | ------------------------------------------------------- | ------------------------------------------------------ |
| age            | Range : 5.0 - 10.3, Mean : 8.1222, Median : 8.15        | Range : 5.0 - 10.3, Mean : 8.4794, Median : 8.6        |
| mass           | Range : 0.1 - 298.5448, Mean : 11.3677, Median : 2.8697 | Range : 0.1 - 298.5448, Mean : 7.3915, Median : 2.0304 |
| log_teff       | Range : 3.34 - 5.3788, Mean : 4.0309, Median : 3.8557   | Range : 3.34 - 5.5871, Mean : 3.8475, Median : 3.6808  |
| log_g          | Range : -1.08 - 7.9405, Mean : 3.515, Median : 3.8557   | Range : -1.14 - 8.6121, Mean : 2.1347, Median : 2.0681 |
| log_R          | Range : -1.86 - 3.1298, Mean : 0.7296, Median : 0.6976  | Range : -2.09 - 3.1298, Mean : 1.3543, Median : 1.4604 |

### T_eff parameter

In [13]:
compare_metrics(path, value_rounding=5,
                output_parameters=["T_eff", "log_g", "radius"],
                physical_models=["MIST"],
                model_names=["random_forest"],
                data_filters=["Base_old"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter   output_parameter                       
random_forest MIST           Base_old T_eff              3.34467 - 5.58714   
                                      log_g               -1.1449 - 8.6121   
                                      radius            -2.08517 - 3.12975   

                                                            RVE     RMSE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.85642  0.14016   
                                      log_g             0.54794  1.32716   
                                      radius            0.55631  0.66360   

                                                            MAE    MedAE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.06697  0.00553   
                                      log_g             0.68127  0.04931   
                                      radius            0.34072  0.02471   

                                                           CORR   MAX_ER  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.92628  1.58571   
                                      log_g             0.75158  7.91597   
                                      radius            0.75665  3.95800   

                                                                                                        Percentiles  \
model         physical_model filter   output_parameter                                                                
random_forest MIST           Base_old T_eff              75 : 0.07494 / 90 : 0.23372 / 95 : 0.35136 / 99 : 0.56158    
                                      log_g              75 : 0.89708 / 90 : 2.51135 / 95 : 3.38312 / 99 : 4.55568    
                                      radius             75 : 0.44863 / 90 : 1.25577 / 95 : 1.69166 / 99 : 2.27713    

                                                             time  
model         physical_model filter   output_parameter             
random_forest MIST           Base_old T_eff             674.38852  
                                      log_g             674.38852  
                                      radius            674.38852

Results for the phase category with a value of 0.0


value range  \
model         physical_model filter   output_parameter                       
random_forest MIST           Base_old T_eff              3.34467 - 4.90341   
                                      log_g             -0.59731 - 5.43309   
                                      radius            -0.99747 - 3.08275   

                                                            RVE     RMSE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.80304  0.17263   
                                      log_g            -4.29496  1.63895   
                                      radius           -0.59010  0.81950   

                                                            MAE    MedAE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.08037  0.00112   
                                      log_g             0.80781  0.00401   
                                      radius            0.40420  0.00259   

                                                           CORR   MAX_ER  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.91864  1.58571   
                                      log_g             0.37716  5.57552   
                                      radius            0.54558  2.78953   

                                                                                                        Percentiles  \
model         physical_model filter   output_parameter                                                                
random_forest MIST           Base_old T_eff              75 : 0.08948 / 90 : 0.31533 / 95 : 0.45813 / 99 : 0.61607    
                                      log_g              75 : 1.01317 / 90 : 3.49465 / 95 : 4.15935 / 99 : 4.80819    
                                      radius             75 : 0.50708 / 90 : 1.74746 / 95 : 2.07953 / 99 : 2.40365    

                                                             time  
model         physical_model filter   output_parameter             
random_forest MIST           Base_old T_eff             674.38852  
                                      log_g             674.38852  
                                      radius            674.38852

Results for the phase category with a value of 2.0


value range  \
model         physical_model filter   output_parameter                       
random_forest MIST           Base_old T_eff              3.42453 - 4.70936   
                                      log_g             -0.32878 - 4.40791   
                                      radius             0.09099 - 2.90038   

                                                            RVE     RMSE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.94674  0.07435   
                                      log_g             0.01791  0.91349   
                                      radius            0.15494  0.45674   

                                                            MAE    MedAE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.03287  0.00312   
                                      log_g             0.44481  0.01621   
                                      radius            0.22241  0.00811   

                                                           CORR   MAX_ER  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.97383  0.74220   
                                      log_g             0.51086  4.41580   
                                      radius            0.56201  2.20794   

                                                                                                        Percentiles  \
model         physical_model filter   output_parameter                                                                
random_forest MIST           Base_old T_eff              75 : 0.03232 / 90 : 0.10811 / 95 : 0.16437 / 99 : 0.33827    
                                      log_g                75 : 0.51838 / 90 : 1.7431 / 95 : 2.3302 / 99 : 3.28588    
                                      radius             75 : 0.25918 / 90 : 0.87149 / 95 : 1.16491 / 99 : 1.64297    

                                                             time  
model         physical_model filter   output_parameter             
random_forest MIST           Base_old T_eff             674.38852  
                                      log_g             674.38852  
                                      radius            674.38852

Results for the phase category with a value of 3.0


value range  \
model         physical_model filter   output_parameter                       
random_forest MIST           Base_old T_eff               3.35268 - 5.3788   
                                      log_g              -1.0829 - 7.94051   
                                      radius            -1.86107 - 3.12975   

                                                            RVE     RMSE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.82481  0.14151   
                                      log_g             0.38245  0.95863   
                                      radius            0.51441  0.47939   

                                                            MAE    MedAE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.07242  0.01368   
                                      log_g             0.54691  0.11312   
                                      radius            0.27352  0.05659   

                                                           CORR   MAX_ER  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.91279  1.00208   
                                      log_g             0.69035  7.50320   
                                      radius            0.74976  3.75159   

                                                                                                        Percentiles  \
model         physical_model filter   output_parameter                                                                
random_forest MIST           Base_old T_eff              75 : 0.09327 / 90 : 0.22966 / 95 : 0.33934 / 99 : 0.55474    
                                      log_g              75 : 0.88332 / 90 : 1.79715 / 95 : 2.23395 / 99 : 3.08305    
                                      radius             75 : 0.44189 / 90 : 0.89859 / 95 : 1.11736 / 99 : 1.54214    

                                                             time  
model         physical_model filter   output_parameter             
random_forest MIST           Base_old T_eff             674.38852  
                                      log_g             674.38852  
                                      radius            674.38852

Results for the phase category with a value of 4.0


value range  \
model         physical_model filter   output_parameter                       
random_forest MIST           Base_old T_eff              3.45594 - 5.58714   
                                      log_g                -0.862 - 8.6121   
                                      radius            -2.08517 - 3.07966   

                                                            RVE     RMSE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.72308  0.12439   
                                      log_g             0.44189  0.78065   
                                      radius            0.61375  0.39029   

                                                            MAE    MedAE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.05577  0.01056   
                                      log_g             0.42883  0.12655   
                                      radius            0.21440  0.06328   

                                                           CORR   MAX_ER  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.86655  0.89689   
                                      log_g             0.71111  6.20091   
                                      radius            0.80118  3.10046   

                                                                                                        Percentiles  \
model         physical_model filter   output_parameter                                                                
random_forest MIST           Base_old T_eff              75 : 0.04696 / 90 : 0.17431 / 95 : 0.30889 / 99 : 0.55789    
                                      log_g               75 : 0.57144 / 90 : 1.3231 / 95 : 1.86109 / 99 : 2.91891    
                                      radius             75 : 0.28573 / 90 : 0.66156 / 95 : 0.93056 / 99 : 1.45945    

                                                             time  
model         physical_model filter   output_parameter             
random_forest MIST           Base_old T_eff             674.38852  
                                      log_g             674.38852  
                                      radius            674.38852

Results for the phase category with a value of 5.0


value range  \
model         physical_model filter   output_parameter                       
random_forest MIST           Base_old T_eff              3.34914 - 5.54353   
                                      log_g              -1.1449 - 8.52744   
                                      radius            -2.05402 - 3.04812   

                                                            RVE     RMSE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.79922  0.13439   
                                      log_g             0.17281  1.38527   
                                      radius            0.28797  0.69264   

                                                            MAE    MedAE  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.07095  0.01224   
                                      log_g             0.77492  0.13864   
                                      radius            0.38745  0.06932   

                                                           CORR   MAX_ER  \
model         physical_model filter   output_parameter                     
random_forest MIST           Base_old T_eff             0.90345  1.46849   
                                      log_g             0.64915  7.91597   
                                      radius            0.68912  3.95800   

                                                                                                        Percentiles  \
model         physical_model filter   output_parameter                                                                
random_forest MIST           Base_old T_eff              75 : 0.09529 / 90 : 0.23667 / 95 : 0.33059 / 99 : 0.49124    
                                      log_g                75 : 1.1798 / 90 : 2.6404 / 95 : 3.29779 / 99 : 4.45991    
                                      radius             75 : 0.58971 / 90 : 1.31996 / 95 : 1.64907 / 99 : 2.23045    

                                                             time  
model         physical_model filter   output_parameter             
random_forest MIST           Base_old T_eff             674.38852  
                                      log_g             674.38852  
                                      radius            674.38852

### log_g parameter

In [ ]:
compare_metrics(path, 
                output_parameters=["log_g"],
                physical_models=["MIST"],
                data_filters=["Base", "Base_old"])

### Radius parameter

In [ ]:
compare_metrics(path, 
                output_parameters=["radius"],
                physical_models=["MIST"],
                data_filters=["Base", "Base_old"])

## Base vs mass_filtered

| Parameter      | Base                                                    | mass_filtered_smaller_3                                | mass_filtered_smaller_14                               | mass_filtered_smaller_30                               |
| -------------  | ------------------------------------------------------- | ------------------------------------------------------ | ------------------------------------------------------ | ------------------------------------------------------ |
| age            | Range : 5.0 - 10.3, Mean : 8.1222, Median : 8.15        | Range : 6.2 - 10.3, Mean : 9.1366, Median : 9.2        | Range : 5.0 - 10.3, Mean : 8.4693, Median : 8.5        | Range : 0.1 - 29.9991, Mean : 5.0208, Median : 2.4047  |
| mass           | Range : 0.1 - 298.5448, Mean : 11.3677, Median : 2.8697 | Range : 0.1 - 3.0, Mean : 1.3172, Median : 1.1821      | Range : 0.1 - 13.9995, Mean : 3.3831, Median : 2.0289  | Range : 0.1 - 3.0, Mean : 1.4429, Median : 1.3194      |
| log_teff       | Range : 3.34 - 5.3788, Mean : 4.0309, Median : 3.8557   | Range : 3.34 - 5.2138, Mean : 3.8022, Median : 3.7554  | Range : 3.34 - 5.2138, Mean : 3.9383, Median : 3.867   | Range : 3.34 - 5.3542, Mean : 3.9845, Median : 3.9191  |
| log_g          | Range : -1.08 - 7.9405, Mean : 3.515, Median : 3.8557   | Range : -1.08 - 7.9405, Mean : 3.6564, Median : 4.1712 | Range : -1.08 - 7.9405, Mean : 3.5574, Median : 3.9537 | Range : -1.08 - 7.9405, Mean : 3.5437, Median : 3.9117 |
| log_R          | Range : -1.86 - 3.1298, Mean : 0.7296, Median : 0.6976  | Range : -1.86 - 2.7813, Mean : 0.4084, Median : 0.2349 | Range : -1.86 - 3.0828, Mean : 0.6007, Median : 0.5357 | Range : -1.86 - 3.1298, Mean : 0.6537, Median : 0.6118 |

### Mass parameter

In [11]:
compare_metrics(path, 
                output_parameters=["mass"],
                physical_models=["MIST"],
                data_filters=["Base", "mass_smaller_3", "mass_smaller_14", "mass_smaller_30"])

Results for the Global category with a value of 1.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
KNN               MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
linear_regression MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
mlp               MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
random_forest     MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
xgboost           MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            mass              0.9951817784001245   
                                 mass_smaller_3  mass              0.9986106418886865   
                                 mass_smaller_14 mass              0.9997601454617543   
                                 mass_smaller_30 mass              0.9991406839290182   
KNN               MIST           Base            mass              0.9895916504979603   
                                 mass_smaller_3  mass              0.9984125750212497   
                                 mass_smaller_14 mass              0.9993586825300925   
                                 mass_smaller_30 mass              0.9987489616665379   
linear_regression MIST           Base            mass              0.3028040512725666   
                                 mass_smaller_3  mass              0.6829393130134978   
                                 mass_smaller_14 mass              0.7182024528881572   
                                 mass_smaller_30 mass              0.63535624

Results for the phase category with a value of 0.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
KNN               MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
linear_regression MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
mlp               MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
random_forest     MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   
xgboost           MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass              0.0999979840073621 - 13.998624815474695   
                                 mass_smaller_30 mass               0.099998052173157 - 29.999136881366123   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            mass              0.9935567994940296   
                                 mass_smaller_3  mass              0.9998109603044399   
                                 mass_smaller_14 mass              0.9997672373377944   
                                 mass_smaller_30 mass              0.9989288712610146   
KNN               MIST           Base            mass              0.9843387655163102   
                                 mass_smaller_3  mass              0.9992443930168096   
                                 mass_smaller_14 mass              0.9991555335313982   
                                 mass_smaller_30 mass              0.9982834222549964   
linear_regression MIST           Base            mass              0.2729320152593433   
                                 mass_smaller_3  mass              0.8226816036408051   
                                 mass_smaller_14 mass               0.697370236171527   
                                 mass_smaller_30 mass              0.59001680

Results for the phase category with a value of 2.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6542770535966311 - 13.927740111688449   
                                 mass_smaller_30 mass               0.6538581961218565 - 29.67887151062485   
KNN               MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6542770535966311 - 13.927740111688449   
                                 mass_smaller_30 mass               0.6538581961218565 - 29.67887151062485   
linear_regression MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6542770535966311 - 13.927740111688449   
                                 mass_smaller_30 mass               0.6538581961218565 - 29.67887151062485   
mlp               MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6542770535966311 - 13.927740111688449   
                                 mass_smaller_30 mass               0.6538581961218565 - 29.67887151062485   
random_forest     MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6542770535966311 - 13.927740111688449   
                                 mass_smaller_30 mass               0.6538581961218565 - 29.67887151062485   
xgboost           MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6542770535966311 - 13.927740111688449   
                                 mass_smaller_30 mass               0.6538581961218565 - 29.67887151062485   

                                                                                   RVE  \
model             physical_model filter          output_parameter                        
decision_tree     MIST           Base            mass               0.9991682912199935   
                                 mass_smaller_3  mass               0.9992078967946935   
                                 mass_smaller_14 mass               0.9999461137434603   
                                 mass_smaller_30 mass               0.9998856763326173   
KNN               MIST           Base            mass               0.9993836364150782   
                                 mass_smaller_3  mass               0.9991876883590323   
                                 mass_smaller_14 mass               0.9997771557894501   
                                 mass_smaller_30 mass               0.9998926681676554   
linear_regression MIST           Base            mass              0.31780480327703364   
                                 mass_smaller_3  mass               0.6094803842541214   
                                 mass_smaller_14 mass                0.762400313993691   
                                 mass_smaller_30 mass           

Results for the phase category with a value of 3.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            mass                0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass               0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass              0.5418086127397849 - 13.996563361499728   
                                 mass_smaller_30 mass               0.5418086438439156 - 29.99464069960494   
KNN               MIST           Base            mass                0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass               0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass              0.5418086127397849 - 13.996563361499728   
                                 mass_smaller_30 mass               0.5418086438439156 - 29.99464069960494   
linear_regression MIST           Base            mass                0.5418086127397849 - 288.814589557117   
                                 mass_smaller_3  mass               0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass              0.5418086127397849 - 13.996563361499728   
                                 mass_smaller_30 mass               0.5418086438439156 - 29.99464069960494   
mlp               MIST           Base            mass                0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass               0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass              0.5418086127397849 - 13.996563361499728   
                                 mass_smaller_30 mass               0.5418086438439156 - 29.99464069960494   
random_forest     MIST           Base            mass                0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass               0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass              0.5418086127397849 - 13.996563361499728   
                                 mass_smaller_30 mass               0.5418086438439156 - 29.99464069960494   
xgboost           MIST           Base            mass                0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass               0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass              0.5418086127397849 - 13.996563361499728   
                                 mass_smaller_30 mass               0.5418086438439156 - 29.99464069960494   

                                                                                   RVE  \
model             physical_model filter          output_parameter                        
decision_tree     MIST           Base            mass               0.9918974335576843   
                                 mass_smaller_3  mass               0.9925088328108644   
                                 mass_smaller_14 mass               0.9994746327197954   
                                 mass_smaller_30 mass               0.9987842548516085   
KNN               MIST           Base            mass               0.9860770182495742   
                                 mass_smaller_3  mass                 0.99350766733889   
                                 mass_smaller_14 mass               0.9993531353657983   
                                 mass_smaller_30 mass               0.9986346706699237   
linear_regression MIST           Base            mass              0.37879056102710573   
                                 mass_smaller_3  mass              0.15358248802528074   
                                 mass_smaller_14 mass               0.7133552017917841   
                                 mass_smaller_30 mass           

Every comparison is proportional to the mass.

Linear regression : 
- mass_smaller_3 : worse results than Base on every metric
- mass_smaller_14 : worse results than Base on every metric
- mass_smaller_30 : worse results than Base on every metric except MAX_ER

Decision tree : 
- mass_smaller_3 : worse results than Base on every metric
- mass_smaller_14 : worse results than Base on MAE, MedAE and all percentiles except P99, better results on RMSE and MAX_ER \
&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; => worse results for typical predictions but better at predictions with large errors
- mass_smaller_30 : worse results than Base on MAE, medAE and all percentiles except P99, better results on RMSE and MAX_ER\
&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; => worse results for typical predictions but better at predictions with large errors

KNN : 
- mass_smaller_3 : worse results than Base on every metric except MAX_ER
- mass_smaller_14 : worse results than Base on MAE, MedAE and all percentiles except P99, better results on RMSE and MAX_ER \
&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; => worse results for typical predictions but better at predictions with large errors
- mass_smaller_30 : worse results than Base on MAE, MedAE and all percentiles except P99, better results on RMSE and MAX_ER \
&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; => worse results for typical predictions but better at predictions with large errors

Random forest : 
- mass_smaller_3 : worse results than Base on every metric
- mass_smaller_14 : worse results than Base on MAE, MedAE and all percentiles except P99, better results on RMSE and MAX_ER \
&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; => worse results for typical predictions but better at predictions with large errors
- mass_smaller_30 : worse results than Base on MAE, MedAE and all percentiles except P99, better results on MAX_ER and similar on RMSE

XGBoost : 
- mass_smaller_3 : worse results than Base on every metric
- mass_smaller_14 : worse results than Base on MAE, MedAE and all percentiles except P99, better results on MAX_ER and similar on RMSE\
&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; => worse results for typical predictions but better at predictions with large errors
- mass_smaller_30 : worse results than Base on every metric except MAX_ER

MLP : 
- mass_smaller_3 : worse results than Base on MAE, MedAE and all percentiles except P99, better results on MAX_ER and similar on RMSE
- mass_smaller_14 : worse results than Base on MAE, MedAE and all percentiles except P99, better results on RMSE and MAX_ER\
&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; => worse results for typical predictions but better at predictions with large errors
- mass_smaller_30 : worse results than Base on every metric except MAX_ER

#### Conclusions on mass filtering

mass_smaller_3 is always worse than Base.

If reducing worst-case errors is the priority, mass_smaller_14 is preferable; if average accuracy matters more, Base is better.

mass_smaller_30 is a middle ground between Base and mass_smaller_14 but it does not offer enough improvements to be interesting to use.

#### Best models for the mass parameter

The three best performing models in order for the mass parameter are Random forest, Decision tree and KNN.

### Radius parameter

In [12]:
compare_metrics(path, 
                output_parameters=["radius"],
                physical_models=["MIST"],
                data_filters=["Base", "mass_smaller_3", "mass_smaller_14", "mass_smaller_30"])

Results for the Global category with a value of 1.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0827540579576227   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
KNN               MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0827540579576227   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
linear_regression MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0827540579576227   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
mlp               MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0827540579576227   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
random_forest     MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0827540579576227   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
xgboost           MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0827540579576227   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            radius            0.9997998092796726   
                                 mass_smaller_3  radius            0.9998692605707555   
                                 mass_smaller_14 radius            0.9998337327080868   
                                 mass_smaller_30 radius            0.9998199780100178   
KNN               MIST           Base            radius            0.9998995813284512   
                                 mass_smaller_3  radius            0.9999075272015158   
                                 mass_smaller_14 radius            0.9999258045438245   
                                 mass_smaller_30 radius             0.999924357563505   
linear_regression MIST           Base            radius            0.9839439743859983   
                                 mass_smaller_3  radius            0.9886074464966265   
                                 mass_smaller_14 radius            0.9896569006470799   
                                 mass_smaller_30 radius            0.98938869

Results for the phase category with a value of 0.0


value range  \
model             physical_model filter          output_parameter                                             
decision_tree     MIST           Base            radius            -0.9974628481553472 - 3.0827540579576227   
                                 mass_smaller_3  radius            -0.9974234436680278 - 0.7629161677552769   
                                 mass_smaller_14 radius            -0.9974747647513328 - 3.0827540579576227   
                                 mass_smaller_30 radius            -0.9974747647513328 - 3.0827540579576227   
KNN               MIST           Base            radius            -0.9974628481553472 - 3.0827540579576227   
                                 mass_smaller_3  radius            -0.9974234436680278 - 0.7629161677552769   
                                 mass_smaller_14 radius            -0.9974747647513328 - 3.0827540579576227   
                                 mass_smaller_30 radius            -0.9974747647513328 - 3.0827540579576227   
linear_regression MIST           Base            radius            -0.9974628481553472 - 3.0827540579576227   
                                 mass_smaller_3  radius            -0.9974234436680278 - 0.7629161677552769   
                                 mass_smaller_14 radius            -0.9974747647513328 - 3.0827540579576227   
                                 mass_smaller_30 radius            -0.9974747647513328 - 3.0827540579576227   
mlp               MIST           Base            radius            -0.9974628481553472 - 3.0827540579576227   
                                 mass_smaller_3  radius            -0.9974234436680278 - 0.7629161677552769   
                                 mass_smaller_14 radius            -0.9974747647513328 - 3.0827540579576227   
                                 mass_smaller_30 radius            -0.9974747647513328 - 3.0827540579576227   
random_forest     MIST           Base            radius            -0.9974628481553472 - 3.0827540579576227   
                                 mass_smaller_3  radius            -0.9974234436680278 - 0.7629161677552769   
                                 mass_smaller_14 radius            -0.9974747647513328 - 3.0827540579576227   
                                 mass_smaller_30 radius            -0.9974747647513328 - 3.0827540579576227   
xgboost           MIST           Base            radius            -0.9974628481553472 - 3.0827540579576227   
                                 mass_smaller_3  radius            -0.9974234436680278 - 0.7629161677552769   
                                 mass_smaller_14 radius            -0.9974747647513328 - 3.0827540579576227   
                                 mass_smaller_30 radius            -0.9974747647513328 - 3.0827540579576227   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            radius            0.9997824748596408   
                                 mass_smaller_3  radius            0.9998320816205213   
                                 mass_smaller_14 radius            0.9997928510771583   
                                 mass_smaller_30 radius            0.9998036006117262   
KNN               MIST           Base            radius            0.9998348784628862   
                                 mass_smaller_3  radius            0.9996595379770277   
                                 mass_smaller_14 radius            0.9998244565962764   
                                 mass_smaller_30 radius            0.9998692256870981   
linear_regression MIST           Base            radius            0.9755626571141772   
                                 mass_smaller_3  radius            0.9597040325540658   
                                 mass_smaller_14 radius            0.9781333879983054   
                                 mass_smaller_30 rad

Results for the phase category with a value of 2.0


value range  \
model             physical_model filter          output_parameter                                           
decision_tree     MIST           Base            radius            0.0919862423867022 - 2.900376516729969   
                                 mass_smaller_3  radius            0.0909920840275404 - 2.353416446984946   
                                 mass_smaller_14 radius            0.0909920840275404 - 2.900376516729969   
                                 mass_smaller_30 radius            0.0909920840275404 - 2.900376516729969   
KNN               MIST           Base            radius            0.0919862423867022 - 2.900376516729969   
                                 mass_smaller_3  radius            0.0909920840275404 - 2.353416446984946   
                                 mass_smaller_14 radius            0.0909920840275404 - 2.900376516729969   
                                 mass_smaller_30 radius            0.0909920840275404 - 2.900376516729969   
linear_regression MIST           Base            radius            0.0919862423867022 - 2.900376516729969   
                                 mass_smaller_3  radius            0.0909920840275404 - 2.353416446984946   
                                 mass_smaller_14 radius            0.0909920840275404 - 2.900376516729969   
                                 mass_smaller_30 radius            0.0909920840275404 - 2.900376516729969   
mlp               MIST           Base            radius            0.0919862423867022 - 2.900376516729969   
                                 mass_smaller_3  radius            0.0909920840275404 - 2.353416446984946   
                                 mass_smaller_14 radius            0.0909920840275404 - 2.900376516729969   
                                 mass_smaller_30 radius            0.0909920840275404 - 2.900376516729969   
random_forest     MIST           Base            radius            0.0919862423867022 - 2.900376516729969   
                                 mass_smaller_3  radius            0.0909920840275404 - 2.353416446984946   
                                 mass_smaller_14 radius            0.0909920840275404 - 2.900376516729969   
                                 mass_smaller_30 radius            0.0909920840275404 - 2.900376516729969   
xgboost           MIST           Base            radius            0.0919862423867022 - 2.900376516729969   
                                 mass_smaller_3  radius            0.0909920840275404 - 2.353416446984946   
                                 mass_smaller_14 radius            0.0909920840275404 - 2.900376516729969   
                                 mass_smaller_30 radius            0.0909920840275404 - 2.900376516729969   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            radius            0.9995892806800809   
                                 mass_smaller_3  radius            0.9995587617625535   
                                 mass_smaller_14 radius            0.9995197001584081   
                                 mass_smaller_30 radius            0.9994948425223688   
KNN               MIST           Base            radius            0.9999061973615614   
                                 mass_smaller_3  radius            0.9998817172665754   
                                 mass_smaller_14 radius            0.9998950136532548   
                                 mass_smaller_30 radius             0.999896732789779   
linear_regression MIST           Base            radius             0.969243149628593   
                                 mass_smaller_3  radius            0.9832984636261459   
                                 mass_smaller_14 radius            0.9821218597814573   
                                 mass_smaller_30 radius            0.9812242174606677   
mlp          

Results for the phase category with a value of 3.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0658785857783406   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
KNN               MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0658785857783406   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
linear_regression MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0658785857783406   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
mlp               MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0658785857783406   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
random_forest     MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0658785857783406   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   
xgboost           MIST           Base            radius            -1.861069007171269 - 3.1297545143214007   
                                 mass_smaller_3  radius             -1.861069007171269 - 2.772287451333455   
                                 mass_smaller_14 radius            -1.859644360427887 - 3.0658785857783406   
                                 mass_smaller_30 radius            -1.861069007171269 - 3.1297545143214007   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            radius            0.9994488214856186   
                                 mass_smaller_3  radius            0.9995610591594069   
                                 mass_smaller_14 radius            0.9995284522938918   
                                 mass_smaller_30 radius             0.999518246344412   
KNN               MIST           Base            radius            0.9997529203147828   
                                 mass_smaller_3  radius            0.9997205248038691   
                                 mass_smaller_14 radius            0.9998098020686446   
                                 mass_smaller_30 radius            0.9998064077445129   
linear_regression MIST           Base            radius            0.9697033500946161   
                                 mass_smaller_3  radius            0.9684680555577966   
                                 mass_smaller_14 radius            0.9740874504101877   
                                 mass_smaller_30 radius            0.97395369

Linear regression : 
- mass_smaller_3 : better/similar results than Base on every metric except MedAE
- mass_smaller_14 : better results than Base on every metric except for MedAE which is similar
- mass_smaller_30 : better results than Base on every metric except for MedAE which is similar, very similar to mass_smaller_14

Decision tree : 
- mass_smaller_3 : better results than Base on every metric
- mass_smaller_14 : better results than Base on every metric but less than mass_smaller_3
- mass_smaller_30 : better or similar results than Base on every metric except MAX_ER, less improvement than mass_smaller_3 and mass_smaller_14

KNN : 
- mass_smaller_3 : similar results than Base except for MAX_ER which is better
- mass_smaller_14 : better results than Base on every metric
- mass_smaller_30 : better results than Base on every metric but less than mass_smaller_14

Random forest : 
- mass_smaller_3 : better results than Base on every metric
- mass_smaller_14 : better results than Base on every metric, less than mass_smaller_3 except for MAX_ER
- mass_smaller_30 : better results than Base on every metric, less than mass_smaller_3 and mass_smaller_14

XGBoost : 
- mass_smaller_3 : better results than Base on every metric
- mass_smaller_14 : better results than Base on every metric except on RMSE and MAX_ER
- mass_smaller_30 : better results than Base on every metric except on MAX_ER

MLP : 
- mass_smaller_3 : way better results than Base on every metric
- mass_smaller_14 : way better results than Base on every metric but less than mass_smaller_3
- mass_smaller_30 : way better results than Base on every metric but less than mass_smaller_3 and mass_smaller_14

#### Conclusions on radius

mass_smaller_3 almost always gives better results than Base.

mass_smaller_14 almost always gives better results than Base but most of the time it improves less than mass_smaller_3. Although it is more consistently better.

mass_smaller_30 almost always gives better results than Base but most of the time it improves less than mass_smaller_3 and mass_smaller_14

#### Best models for the radius parameter

The three best performing models in order for the radius parameter are Random forest, KNN and Decision tree.

### Conclusion on both

It seems that using the filtering mass_smaller_14 gives better results for radius and gives less high errors for the mass.

This filtering might be the best choice as it is also the range of values for PARSEC which would make comparing models between the two easier.